rules/do-you-know-how-to-use-connection-strings/rule.md

```diff
@@ -10,172 +10,38 @@ authors:
-            })
-            ).SingleOrDefault();
-    }
-}
-```
-
-::: bad
-Figure: Bad Example - Option #1 Connection strings do not belong in your code, anyone seeing this could access your database
-:::
-
-#### Option #2 - Using connection strings in `appsettings.json`
-
-```js
-// In appsettings.json
-{
-  "ApplicationSecrets": {
-    "LicenceKey": "ABCD-1234-HIJK",
-    "ConnectionString": "Server=localhost;Database=Northwind;Trusted_connection=false;user id=sa;pwd=admin"
-  }
-}
-```
-
-::: bad
-Figure: Bad Example - Option #2 Connection strings do not belong in your `appsettings.json` either, once committed to version control they are hard to remove
-:::
-
-#### Option #3 - Referencing loosely typed connection strings
-
-```cs
-public class MyDataService
-{
-    public readonly string _connectionString;
-    
-    public MyDataService(IConfiguration config)
-    {
-        // In Production, your connection string will be read from Key Vault instead of appsettings.json
-	// Note: we are grabbing the connection string setting directly from config as a string using another magic string - this is bad!
-        _connectionString = config["ApplicationSecrets:SqlConnectionString"];
-    }
-    
-    private async Task<string> GetCustomerDetails(CustomerDetailsQuery request)
-    {
-        var sql = @"SELECT * FROM CustomerDetails WHERE CustomerId = @auctionId";
-        await using var db = new SqlConnection(_connectionString);
-        return (await db.QueryAsync<string>(sql,
-            new
-            {
-                customerId = request.CustomerId
-            })
-            ).SingleOrDefault();
-    }
-}
-```
-
-::: bad
-Figure: Bad Example - Option #3 Referencing a loosely typed connection string defined in application settings
-:::
-
-#### Option #4 - User Secrets
-
-An alternative to putting secrets into `appsettings.json` is via [User Secrets](https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-5.0&tabs=windows)
-
-User secrets provide you with a secret file (called `secrets.json`) that is stored on your local machine away from the current code repository and cannot be committed by accident.
-
-To access your User Secrets:
-
-1. Right-click your project in Visual Studio
-2. Select Manage User Secrets
-3. Edit the content of your `secrets.json` file
-4. Save the file
-
-![Figure: OK Example - Option #4 User secrets (secrets.json)](user-secrets.png)
+#### Option #2 - Connection Strings with passwords or keys
 
-```cs
-// In ApplicationSecrets.cs
-public class ApplicationSecrets
-{
-    public string SqlConnectionString { get; set; }
-    public string LicenseKey { get; set; }
-}
-```
+If you have to use some sort of secret or key to login to the service being referenced, then some thought needs to be given to how those secrets can be secured.
+Take a look at https://www.ssw.com.au/rules/store-your-secrets-securely to learn how to keep your secrets secure.
```

Comment:
Change from:
Take a look at https://www.ssw.com.au/rules/store-your-secrets-securely to learn how to keep your secrets secure.

To:
Take a look at [Do you store your secrets securely?](https://www.ssw.com.au/rules/store-your-secrets-securely) to learn how to keep your secrets secure

Reply:
I agree that your suggestion is better, but I believe the standard is to have fully qualified URLs in the rules